pam/internal/adapter/gdmmodel_uimodel_test.go

```diff
@@ -0,0 +1,176 @@
+package adapter
+
+import (
+	"context"
+	"reflect"
+	"slices"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/internal/log"
+	"github.com/ubuntu/authd/pam/internal/gdm"
+	"github.com/ubuntu/authd/pam/internal/gdm_test"
+	"github.com/ubuntu/authd/pam/internal/proto"
+)
+
+// gdmTestUIModel is an override of [UIModel] used for testing the module with gdm.
+type gdmTestUIModel struct {
+	UIModel
+
+	mu sync.Mutex
+
+	gdmHandler *gdmConvHandler
+
+	wantMessages        []tea.Msg
+	wantMessagesHandled chan struct{}
+
+	program           *tea.Program
+	programHasQuit    atomic.Bool
+	programShouldQuit atomic.Bool
```

Comment:
Oh, I actually noticed that `programHasQuit` is not used anymore so I've dropped it.

As per `programShouldQuit`, it's used also without locks, since it's used in the "control" goroutine before we leave it and we don't have any lock there, nor we should have it.

So in such case we need this.